src/super_gradients/compliance/check_compliance.py

```diff
@@ -0,0 +1,122 @@
+    """Read requirement.txt from the root, and split it as a list of libs/version"""
+    with open(Path(__file__).parent.parent.parent.parent / "requirements.txt", "r") as f:
+        requirements_str = f.read()
+    return requirements_str.split("\n")
+
+
+def get_installed_libs_with_version() -> Dict[str, str]:
+    """Get all the installed libraries, and outputs it as a dict: lib -> version"""
+    installed_libs_with_version = {}
+    for lib_with_version in freeze():
+        if "==" in lib_with_version:
+            lib, version = lib_with_version.split("==")
+            installed_libs_with_version[lib.lower()] = version
+    return installed_libs_with_version
+
+
+def verify_installed_libraries() -> List[str]:
+    """Check that all installed libs respect the requirement.txt"""
+    requirements = get_libs_requirements()
+    installed_libs_with_version = get_installed_libs_with_version()
+
+    errors = []
+    for requirement in requirements:
+        if ">=" in requirement:
+            constraint = ">="
+        elif "~=" in requirement:
+            constraint = "~="
+        elif "==" in requirement:
+            constraint = "=="
+        else:
+            continue
+
+        lib, required_version_str = requirement.split(constraint)
+
+        if lib.lower() not in installed_libs_with_version.keys():
+            errors.append(f"{lib} required but not found")
+            continue
+
+        installed_version_str = installed_libs_with_version[lib.lower()]
+        installed_version, required_version = Version(installed_version_str), Version(required_version_str)
+
+        is_constraint_respected = {
+            ">=": installed_version >= required_version,
+            "~=": installed_version.major == required_version.major and installed_version.minor == required_version.minor and installed_version.micro >= required_version.micro,
+            "==": installed_version == required_version
+        }
+        if not is_constraint_respected[constraint]:
+            errors.append(
+                f"{lib} is installed with version {installed_version} which does not satisfy {requirement}")
+    return errors
+
+
+def print_error(component_name: str, error: str) -> None:
+    error_message = f"Failed to verify {component_name}: {error}"
+    logger.error(error_message)
+
+
+def check_compliance() -> None:
+    """Run all the compliance test and log everything that does not meet requirements"""
+
+    requirement_checkers = {
+        'operating_system': verify_os,
+        'libraries': verify_installed_libraries,
+    }
+
+    compliance_errors = {}
+    logger.info('Deci Compliance Check Started')
+    logger.info(f'Checking the following components: {list(requirement_checkers.keys())}')
+    logger.info('_' * 20)
+    for test_name, test_function in requirement_checkers.items():
+        logger.info(f"Verifying {test_name}...")
+        try:
+            errors = test_function()
+            if errors:
```

Comment:
but the tests always return a list... this will always be true?

Reply:
I admit I did not know that. but to keep a clean readable code, please make it `if len(errors) > 0`